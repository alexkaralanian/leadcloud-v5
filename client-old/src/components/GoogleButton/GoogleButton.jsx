import React from "react";
import PropTypes from "prop-types";
import { googleButton, googleButton_text } from "./styles.css";

const GoogleButton = () => (
  <button onClick={() => console.log("HI")} className={googleButton}>
    <a target="_self" href="/api/auth/google">
      <span className={googleButton_text}>Login with Google</span>
    </a>
  </button>
);

export default GoogleButton;
