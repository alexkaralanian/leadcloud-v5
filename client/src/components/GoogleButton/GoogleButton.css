.googleButton {
  display: block;
  margin-top: 2.5rem;
  width: 22.5rem;
  height: 4.5rem;
  border-radius: 3rem;
  margin-bottom: 14px;
  outline: none;
  transition: all 0.2s;
  background-color: white;
  border: none;
}

.googleButton a {
  font-size: 14px;
  font-weight: 400;
  text-decoration: none;
}

.googleButton_text {
  color: black;
  font-size: 1.6rem;
  font-weight: 300;
  letter-spacing: 0.125rem;
  margin-right: -0.125rem;
  text-decoration: none;
}

.googleButton:hover {
  transform: translateY(-0.2rem);
  box-shadow: 0 1rem 2rem rgba(0, 0, 0, 0.2);
}

.googleButton:active {
  transform: translateY(-0.1rem);
  box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.2);
}
