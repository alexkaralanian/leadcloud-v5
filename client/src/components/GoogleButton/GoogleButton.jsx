import React from "react";
import "./GoogleButton.css";

const GoogleButton = () => (
  <button className="googleButton">
    <a target="_self" href="/api/auth/google">
      <span className="googleButton_text">Login with Google</span>
    </a>
  </button>
);

export default GoogleButton;
