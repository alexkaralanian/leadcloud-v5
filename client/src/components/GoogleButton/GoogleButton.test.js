import React from "react";
import { shallow, configure } from "enzyme";
import GoogleButton from "./GoogleButton";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

test("Google Button Component renders correctly", () => {
  const component = shallow(<GoogleButton />);
  expect(component).toMatchSnapshot();
});
