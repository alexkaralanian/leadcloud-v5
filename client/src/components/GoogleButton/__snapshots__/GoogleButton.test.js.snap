// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Google Button Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GoogleButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        href="/api/auth/google"
        target="_self"
>
        <span
                className="googleButton_text"
        >
                Login with Google
        </span>
</a>,
      "className": "googleButton",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="googleButton_text"
>
          Login with Google
</span>,
        "href": "/api/auth/google",
        "target": "_self",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login with Google",
          "className": "googleButton_text",
        },
        "ref": null,
        "rendered": "Login with Google",
        "type": "span",
      },
      "type": "a",
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          href="/api/auth/google"
          target="_self"
>
          <span
                    className="googleButton_text"
          >
                    Login with Google
          </span>
</a>,
        "className": "googleButton",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span
            className="googleButton_text"
>
            Login with Google
</span>,
          "href": "/api/auth/google",
          "target": "_self",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login with Google",
            "className": "googleButton_text",
          },
          "ref": null,
          "rendered": "Login with Google",
          "type": "span",
        },
        "type": "a",
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
